packages/firebase_ml_vision/lib/src/text_recognizer.dart

```diff
@@ -20,30 +26,43 @@ part of firebase_ml_vision;
 /// ```
 class TextRecognizer {
   TextRecognizer._({
+    CloudTextRecognizerOptions options,
     @required this.modelType,
     @required int handle,
-  })  : _handle = handle,
-        assert(modelType != null);
+  })  : _options = options,
+        _handle = handle,
+        assert(modelType != null),
+        assert((modelType == ModelType.cloud && options != null) ||
+            (modelType == ModelType.onDevice && options == null));
```

Comment:
I first thought this is a bug to be `!= null` and `==null`, then I realized the option if for cloud, maybe name this `cloudRecognizerOptions`?

Reply:
The name was chosen to keep consistency with ImageLabeler. However, it can be changed in favour of clarity.

Reply:
Yeah I think adding cloud is better as this option isn't used in onDevice models at all, so it could be confusing to those users